### PR TITLE
upgrading storybook dependencies from 6.3.9 to 6.5.12

### LINKED ISSUE #800: [chore] Upgrade to latest Storybook
**Is your feature request related to a problem? Please describe.**
Check changelog and upgrade storybook to latest version

### PR DESCRIPTION
**Pull request by dennybiasiolli**
no breaking changes, details here:
https://github.com/storybookjs/storybook/blob/main/CHANGELOG.md#6512-september-14-2022

Close #800

```
yarn add -D \
    @storybook/addon-a11y@^6.5.12 \
    @storybook/addon-essentials@^6.5.12 \
    @storybook/addon-links@^6.5.12 \
    @storybook/addon-storyshots@^6.5.12 \
    @storybook/addons@^6.5.12 \
    @storybook/react@^6.5.12
```

#### PR Checklist
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The unit tests pass locally with my changes (if applicable).
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable).
- [ ] I have added necessary documentation (if appropriate).

### CODE REVIEW

**Comment by dennybiasiolli** [comment]
Rebasing...

File: package.json [hunk 1]
```diff
@@ -116,6 +116,7 @@
     "jest-transform-stub": "^2.0.0",
     "lint-staged": "^12.1.7",
     "postcss-flexbugs-fixes": "^4.2.1",
+    "prettier": "^2.7.1",
```
Comment (dej611) [inline package.json:119]: Remove this

**Comment by dej611** [comment]
Checked this PR and tried to reproduce the prettier bug locally, found a fix:

```
yarn add -D  eslint-plugin-mdx@1.17.1 eslint-plugin-prettier@4.2.1
```

Then rather than declaring the `prettier` dependency directly, use the `yarn overrides`, adding this at the bottom of the `package.json`:

```json
   ...,
   "resolutions": {
    "**/tsdx/jest": "27.5.1",
    "**/tsdx/ts-jest": "27.1.5",
    "**/tsdx/eslint": "^7.32.0",
    "**/tsdx/prettier": "^2.7.1"
  }
```

Next update the `eslintrc.js` file to make the mdx extension work nicely with the imports (there's a bug for the v.1.x of the plugin):
```json
  ...
  "overrides": [
      {
        "files": ['*.mdx'],
        "extends": 'plugin:mdx/recommended',
        "settings": {
          "mdx/code-blocks": false,
        },
        "parserOptions": {  // <= add this
          "ecmaVersion": 12
        },
      }
  ],
  ...
```

Now do a `yarn lint --fix` pass and everything should be gone

**Review by dej611 — CHANGES_REQUESTED** [review]

**Comment by dej611** [comment]
@dennybiasiolli still interested to work on this?

**Comment by dennybiasiolli** [comment]
@dej611 yes, I will work on that this weekend, I got stuck on another task

**Comment by dej611** [comment]
Nice :)

File: package.json [hunk 2]
```diff
@@ -157,5 +157,11 @@
       "node",
       "mdx"
     ]
+  },
+  "resolutions": {
+    "**/tsdx/jest": "27.5.1",
+    "**/tsdx/ts-jest": "27.1.5",
```
Comment (dej611) [inline package.json:163]: Remove these two lines:

```suggestion
```

File: package.json [hunk 3]
```diff
@@ -108,8 +108,8 @@
     "browserslist-config-design-italia": "^1.0.0",
     "chromatic": "^6.3.4",
     "circular-dependency-plugin": "^5.2.2",
-    "eslint-plugin-mdx": "^1.13.0",
-    "eslint-plugin-prettier": "^3.4.0",
+    "eslint-plugin-mdx": "1.17.1",
+    "eslint-plugin-prettier": "^4.2.1",
     "husky": "^7.0.4",
     "identity-obj-proxy": "^3.0.0",
     "jest": "^26",
```
Comment (dej611) [inline package.json:115]: ### Jest changes

To make test pass it requires a little bit more effort than planned.
Resolutions didn't work well for the `jest` upgrade, so we have to override it here:

```suggestion
    "jest": "27.5.1"
```

And also the `ts-jest` below:

```
"ts-jest": "^27.1.5",
```

and add `"babel-jest"` as newer dep:

```
"babel-jest": "^27.5.1",
```

In the script section replace `tsdx test` with `npx jest` for both `"test"` and `"test:ci"` scripts.

In the "jest" configuration add `jsdom` as environment (it was a breaking change in jest@27):
```
"testEnvironment": "jsdom",
```

In the `jest-transformer.js` file add a `.default` to the `babel-jest` require:

```
const babelJest = require('babel-jest').default;
```

**Review by dej611 — COMMENTED** [review]
I had a look on why the tests were failing and found out the resolution route for the jest upgrade wasn't working properly. I've managed to fix this PR locally and added few comments here on how to fix it. Let me know if it works

**Comment by dej611** [comment]
@dennybiasiolli you still working on this?
Let me know if I have to take over.

**Comment by dennybiasiolli** [comment]
@dej611 I'm stuck at work tasks, I think I won't have time to check on this for at least another week.
Feel free to take over this PR if you can work on it before 😉

**Comment by astagi** [comment]
@dennybiasiolli sorry my branch swap (`master` -> `main`) accidentally closed this PR. Now Storybook 7.6.4 is available, if you'd like to contribute feel free to open a PR to upgrade Storybook to the latest version.

**Pull request closed without merging**